test/jdk/java/awt/print/PrinterJob/PrintTranslatedFont.java

```diff
@@ -21,237 +21,127 @@
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+
+import javax.swing.JButton;
+import javax.swing.JOptionPane;
+
+/*
  * @test
  * @bug 6359734
  * @key printer
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary Test that fonts with a translation print where they should.
  * @run main/manual PrintTranslatedFont
  */
+public class PrintTranslatedFont extends Frame {
+    private static final String INSTRUCTIONS =
+            "This test should print a page which contains the same\n" +
+            "content as the test window on the screen, in particular the lines\n" +
+            "should be immediately under the text\n" +
+            "You should also monitor the command line to see if any exceptions\n" +
+            "were thrown\n" +
+            "If an exception is thrown, or the page doesn't print properly\n" +
+            "then the test fails";
+
+    public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(PrintTranslatedFont::new)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build()
+                .awaitAndCheck();
+    }
 
-
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.geom.*;
-import java.awt.print.*;
-import java.text.*;
-
-public class PrintTranslatedFont extends Frame implements ActionListener {
-
- private TextCanvas c;
-
- public static void main(String args[]) {
-
-  String[] instructions =
-        {
-         "You must have a printer available to perform this test",
-         "This test should print a page which contains the same",
-         "content as the test window on the screen, in particular the lines",
-         "should be immediately under the text",
-         "You should also monitor the command line to see if any exceptions",
-         "were thrown",
-         "If an exception is thrown, or the page doesn't print properly",
-         "then the test fails",
-       };
-      Sysout.createDialog( );
-      Sysout.printInstructions( instructions );
-
-    PrintTranslatedFont f = new PrintTranslatedFont();
-    f.show();
- }
-
- public PrintTranslatedFont() {
-    super("JDK 1.2 drawString Printing");
-
-    c = new TextCanvas();
-    add("Center", c);
-
-    Button printButton = new Button("Print");
-    printButton.addActionListener(this);
-    add("South", printButton);
-
-    addWindowListener(new WindowAdapter() {
-       public void windowClosing(WindowEvent e) {
-             System.exit(0);
+    public PrintTranslatedFont() {
+        super("PrintTranslatedFont");
+
+        TextCanvas c = new TextCanvas();
+        add("Center", c);
+
+        JButton b = new JButton("Print");
```

Comment:
```suggestion
        Button b = new Button("Print");
```
It is better not to mix AWT and Swing components; you use `Frame` as the host instead of `JFrame` in this case.